src/py/flwr/cli/flower_toml.py

```diff
@@ -0,0 +1,153 @@
+    """Load flower.toml and return as dict."""
+    if path is None:
+        cur_dir = os.getcwd()
+        toml_path = os.path.join(cur_dir, "flower.toml")
+    else:
+        toml_path = path
+
+    if not os.path.isfile(toml_path):
+        return None
+
+    with open(toml_path, encoding="utf-8") as toml_file:
+        data = tomli.loads(toml_file.read())
+        return data
+
+
+def validate_flower_toml_fields(config: Dict[str, Any]) -> Tuple[bool, List[str]]:
+    """Validate flower.toml fields."""
+    invalid_reasons = []
+
+    if "project" not in config:
+        invalid_reasons.append("Missing [project] section")
+    else:
+        if "name" not in config["project"]:
+            invalid_reasons.append('Property "name" missing in [project]')
+        if "version" not in config["project"]:
+            invalid_reasons.append('Property "version" missing in [project]')
+        if "description" not in config["project"]:
+            invalid_reasons.append('Property "description" missing in [project]')
+        if "license" not in config["project"]:
+            invalid_reasons.append('Property "license" missing in [project]')
+        if "authors" not in config["project"]:
+            invalid_reasons.append('Property "authors" missing in [project]')
+
+    if "flower" not in config:
+        invalid_reasons.append("Missing [flower] section")
+    elif "components" not in config["flower"]:
+        invalid_reasons.append("Missing [flower.components] section")
+    else:
+        if "serverapp" not in config["flower"]["components"]:
+            invalid_reasons.append(
+                'Property "serverapp" missing in [flower.components]'
+            )
+        if "clientapp" not in config["flower"]["components"]:
+            invalid_reasons.append(
+                'Property "clientapp" missing in [flower.components]'
+            )
+
+    return len(invalid_reasons) == 0, invalid_reasons
+
+
+def validate_object_reference(ref: str) -> Tuple[bool, Optional[str]]:
+    """Validate object reference.
+
+    Returns
+    -------
+    tuple(bool, Optional[str]): is_valid, reason why it might not be
+    """
+    module_str, _, attributes_str = ref.partition(":")
+    if not module_str:
+        return (
+            False,
+            f"Missing module in {ref}",
+        )
+    if not attributes_str:
+        return (
+            False,
+            f"Missing attribute in {ref}",
+        )
+
+    # Load module
+    try:
+        module = importlib.import_module(module_str)
+    except ModuleNotFoundError:
+        return False, f"Unable to load module {module_str}"
+
+    # Recursively load attribute
+    attribute = module
+    try:
+        for attribute_str in attributes_str.split("."):
+            attribute = getattr(attribute, attribute_str)
+    except AttributeError:
+        return (
+            False,
+            f"Unable to load attribute {attributes_str} from module {module_str}",
+        )
```

Comment:
Should we attempt to load the module here? This could take a long time in larger (or poorly coded) projects.

If we keep it, we should wrap it in logs that tell the user that something is happening in the background and that it might take a while until the next output is shown.

Reply:
Good point. I'll look into it.

Reply:
@danieljanes I added some prints to the `run.py` which will inform the user about things that might take longer.